naga/src/front/wgsl/parse/word.rs

```diff
@@ -0,0 +1,97 @@
+use std::fmt;
+use std::sync::OnceLock;
+
+use crate::FastHashMap;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Word<'a> {
+    Ident(&'a str),
+    Underscore,
+    Alias,
+    Break,
+    Case,
+    Const,
+    ConstAssert,
+    Continue,
+    Continuing,
+    Default,
+    Discard,
+    Else,
+    Fn,
+    For,
+    If,
+    Let,
+    Loop,
+    Override,
+    Return,
+    Struct,
+    Switch,
+    Var,
+    While,
```

Comment:
Should we add `diagnostic`, `enable` and `requires` here as well? Those are the only ones missing from [Keyword Summary](https://gpuweb.github.io/gpuweb/wgsl/#keyword-summary). `DirectiveKind::from_ident` can then get the `Word` enum instead of an `str`.

Reply:
Yes, that was going to be a follow-up.